### PR TITLE
added  Common UICollectionView config in category

### PR DESCRIPTION
**Pull request by TheCodeTalker**
## Changes in this pull request

Issue fixed: #846 

### Checklist

- [ ] All tests pass. Demo project builds and runs.
- [ ] I added tests, an experiment, or detailed why my change isn't tested.
- [ ] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [ ] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: Source/UICollectionView+IGListKit.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <UIKit/UIKit.h>
+
+@interface UICollectionView (IGListKit)
+/**
+ A method used to configure collectionView background and disable prefetching so cells are configured as they come on screen.
```
Comment (rnystrom) [inline Source/UICollectionView+IGListKit.h]: I should setup a linter, but I try to keep our header docs to 120 characters and then newlining. Mind breaking this into another line?

Also we can remove implementation details from the description, I think. That'll put it under 120 chars.

> Configure the UIAppearance of all UICollectionViews optimized for use with IGListKit.

File: Example/IGListKitExamples/ViewControllers/SingleSectionViewController.swift [hunk 2]
```diff
@@ -76,7 +76,7 @@ final class SingleSectionViewController: UIViewController, IGListAdapterDataSour
     // MARK: - IGListSingleSectionControllerDelegate
     
     func didSelect(_ sectionController: IGListSingleSectionController) {
-        let section = adapter.section(for: sectionController) + 1
+        let section = adapter.section(for: sectionController) + UInt(1)
```
Comment (rnystrom) [inline Example/IGListKitExamples/ViewControllers/SingleSectionViewController.swift]: Is this required? Was the build broken?

Comment (TheCodeTalker) [inline Example/IGListKitExamples/ViewControllers/SingleSectionViewController.swift]: yes, this is required to build. Adapter section return UInt  .

Comment (rnystrom) [inline Example/IGListKitExamples/ViewControllers/SingleSectionViewController.swift]: Are you working off of an old revision? It returns an `NSInteger`

https://github.com/Instagram/IGListKit/blob/master/Source/IGListAdapter.h#L157

**Comment by rnystrom** [comment]
@TheCodeTalker on your branch you're going to need to rebase onto master. It might be best to squash your commits into one first.

```
$ git rebase -i HEAD~4
```

Then squash each of your commits (except the head)

```
pick 1234 added Common UICollectionView config in category
squash 1234 Minor fix for Compile Error In Examples
squash 1234 Update IGListKit.h
squash 1234 Update UICollectionView+IGListKit.h
```

Then you'll want to `git pull origin master`, fix merge conflicts, and make sure your commit is at the tip of everything (`git log`, should be at the top).

After that, you can take care of the `UIAppearance` and unsigned int stuff.

**Comment by TheCodeTalker** [comment]
hey @rnystrom squash into one but  when i do this pull 

apples-MacBook-Pro-3:IGListKit iThink$ git pull upstream master
remote: Counting objects: 4413, done.
remote: Compressing objects: 100% (52/52), done.
remote: Total 4413 (delta 1168), reused 1177 (delta 1157), pack-reused 3204
Receiving objects: 100% (4413/4413), 2.38 MiB | 439.00 KiB/s, done.
Resolving deltas: 100% (3406/3406), completed with 166 local objects.
From https://github.com/Instagram/IGListKit
 * branch            master     -> FETCH_HEAD
 * [new branch]      master     -> upstream/master
Removing docs/Classes/IGListCollectionView.html
Auto-merging docs/Classes/IGListBindingSectionController.html
Removing build_docs.sh
Auto-merging Tests/Objects/IGTestStoryboardSupplementaryView.m
Removing Tests/IGListStackItemControllerTests.m
Auto-merging Tests/IGListSingleSectionControllerTests.m
Auto-merging Tests/IGListSectionMapTests.m
Removing Tests/IGListIndexResultTests.m
Removing Source/NSObject+IGListDiffable.h
Auto-merging Source/Internal/UICollectionView+DebugDescription.h
Removing Source/Internal/NSIndexSet+PrettyDescription.m
Removing Source/IGListSectionType.h
Auto-merging Source/IGListKit.h
CONFLICT (content): Merge conflict in Source/IGListKit.h
Removing Source/IGListIndexSetResult.h
Removing Source/IGListIndexPathResult.h
Removing Source/IGListDiffable.h
Removing Source/IGListCollectionView.m
Removing Source/IGListCollectionView.h
Removing Source/IGListBatchUpdateData.h
Auto-merging Source/Common/Internal/IGListMoveIndexInternal.h
Auto-merging Source/Common/Internal/IGListIndexSetResultInternal.h
Auto-merging Source/Common/Internal/IGListIndexPathResultInternal.h
Auto-merging Source/Common/IGListMoveIndexPath.h
Auto-merging Source/Common/IGListMoveIndex.m
Auto-merging Source/Common/IGListMoveIndex.h
Auto-merging Source/Common/IGListMacros.h
Auto-merging Source/Common/IGListIndexSetResult.m
Auto-merging Source/Common/IGListIndexPathResult.m
Auto-merging Source/Common/IGListExperiments.h
Auto-merging Source/Common/IGListDiff.mm
Auto-merging Source/Common/IGListDiff.h
Auto-merging Source/Common/IGListBatchUpdateData.mm
Auto-merging Source/Common/IGListAssert.h
Removing Pods/Target Support Files/OCMock/OCMock-prefix.pch
Removing Pods/Target Support Files/OCMock/OCMock-dummy.m
Auto-merging Pods/Target Support Files/OCMock-iOS/OCMock-iOS.xcconfig
Auto-merging Pods/Target Support Files/OCMock-iOS/OCMock-iOS.modulemap
Auto-merging Pods/Target Support Files/OCMock-iOS/OCMock-iOS-umbrella.h
Auto-merging IGListKit.xcodeproj/project.pbxproj
CONFLICT (content): Merge conflict in IGListKit.xcodeproj/project.pbxproj
Removing Gemfile.lock
Removing Gemfile
Auto-merging Examples/Examples-tvOS/Pods/Target Support Files/IGListKit/IGListKit.xcconfig
Auto-merging Examples/Examples-tvOS/Pods/Target Support Files/IGListKit/IGListKit-umbrella.h
Auto-merging Examples/Examples-tvOS/IGListKitExamples/Views/LabelCell.swift
Auto-merging Examples/Examples-tvOS/IGListKitExamples/ViewControllers/DemosViewController.swift
Auto-merging Examples/Examples-tvOS/IGListKitExamples/SectionControllers/LabelSectionController.swift
Auto-merging Examples/Examples-tvOS/IGListKitExamples/AppDelegate.swift
Auto-merging Examples/Examples-macOS/Pods/Target Support Files/Pods-IGListKitExamples/Pods-IGListKitExamples-resources.sh
Auto-merging Examples/Examples-macOS/Pods/Target Support Files/IGListKit/Info.plist
Auto-merging Examples/Examples-iOS/Pods/Target Support Files/Pods-IGListKitExamples/Pods-IGListKitExamples.release.xcconfig
Auto-merging Examples/Examples-iOS/Pods/Target Support Files/Pods-IGListKitExamples/Pods-IGListKitExamples.debug.xcconfig
Auto-merging Examples/Examples-iOS/Pods/Target Support Files/Pods-IGListKitExamples/Pods-IGListKitExamples-umbrella.h
Auto-merging Examples/Examples-iOS/Pods/Target Support Files/Pods-IGListKitExamples/Pods-IGListKitExamples-frameworks.sh
Auto-merging Examples/Examples-iOS/IGListKitExamples/Views/SpinnerCell.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/Views/SearchCell.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/Views/RemoveCell.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/Views/NibCell.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/Views/EmbeddedCollectionViewCell.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/Views/DetailLabelCell.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/Views/CenterLabelCell.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/ViewControllers/SingleSectionViewController.swift
CONFLICT (content): Merge conflict in Examples/Examples-iOS/IGListKitExamples/ViewControllers/SingleSectionViewController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/ViewControllers/SearchViewController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/ViewControllers/NestedAdapterViewController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/ViewControllers/MixedDataViewController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/ViewControllers/LoadMoreViewController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/ViewControllers/EmptyViewController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/SectionControllers/UserSectionController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/SectionControllers/SearchSectionController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/SectionControllers/RemoveSectionController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/SectionControllers/HorizontalSectionController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/SectionControllers/GridSectionController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/SectionControllers/ExpandableSectionController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/SectionControllers/EmbeddedSectionController.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/Models/User.swift
Auto-merging Examples/Examples-iOS/IGListKitExamples/Info.plist
Auto-merging Examples/Examples-iOS/IGListKitExamples/Assets.xcassets/AppIcon.appiconset/Contents.json
Removing Example/Pods/Target Support Files/IGListKit/IGListKit-prefix.pch
Removing Example/Pods/Pods.xcodeproj/project.pbxproj
Removing Example/Pods/Manifest.lock
Removing Example/Pods/Local Podspecs/IGListKit.podspec.json
Removing Example/Podfile.lock
Removing Example/Podfile
Removing Example/IGListKitExamples/SectionControllers/DemoSectionController.swift
Removing Example/IGListKitExamples.xcodeproj/project.pbxproj
Auto-merging .github/CONTRIBUTING.md
Automatic merge failed; fix conflicts and then commit the result.


Because of removed Pods i am not able build for fix the conflicts what i have to do??

**Comment by rnystrom** [comment]
@TheCodeTalker you honestly might just want to check out a new branch off of latest and copy/paste your changes there. Looks pretty behind.

**Comment by TheCodeTalker** [comment]
hey @rnystrom i add category i have seeing that 
` [UICollectionView appearance].backgroundColor  = [UIColor whiteColor];
          if (NSFoundationVersionNumber >= NSFoundationVersionNumber_iOS_9_x_Max){
                  [[UICollectionView appearance] setPrefetchingEnabled:NO];
    }
`
is not effecting the UICollectionview PrefetchingEnabled properties why thats so ??.

**Comment by rnystrom** [comment]
cc @jessesquires, just confirmed that `UIAppearance` doesn't actually change anything on the collection view... we should update our guide and remove that line since its misleading. Thankfully in IG we explicitly disable prefetching in all of our collection views, but still.

I wonder if the `IGListAdapter` should just do this automatically?

**Comment by jessesquires** [comment]
@rnystrom oh wat. 😳 that's weird. AFAIK, we used that `UIAppearance` hack in IG. (i wrote it lol) and it *was* working, unless I was drunk. which I probably was. 😉

**Comment by rnystrom** [comment]
@jessesquires tested in IG w/ Xcode 8.3.3 and after init'ing a `UICollectionView`, its `isPrefetchingEnabled` is `YES`.

**Comment by svara** [comment]
@rnystrom IMO, IGListAdapter should have prefetching disabled by default.

**Comment by rnystrom** [comment]
@svara I think I agree too

**Comment by rnystrom** [comment]
I'm going to follow up w/ a quick internal fix for this setting `prefetchingEnabled = NO` internally and documenting this behavior.

**Pull request closed without merging**